lib/heimdall_tools/asff_mapper.rb

```diff
@@ -0,0 +1,230 @@
+        keywords = { encode: method(:encode) }
+        keywords = keywords.merge(@supporting_docs[PRODUCT_ARN_MAPPING[arn || product]]) if @supporting_docs.member?(PRODUCT_ARN_MAPPING[arn || product])
+        PRODUCT_ARN_MAPPING[arn || product].send(func, data, **keywords)
+      elsif default.is_a? Proc
+        default.call
+      else
+        default
+      end
+    end
+
+    def nist_tag(finding)
+      entries = external_product_handler(finding['ProductArn'], finding, :finding_nist_tag, {})
+      tags = entries.map { |rule| rule[:nistid].split('|') }
+      tags.empty? ? DEFAULT_NIST_TAG : tags.flatten.uniq
+    end
+
+    def impact(finding)
+      # there can be findings listed that are intentionally ignored due to the underlying control being superceded by a control from a different standard
+      if finding.member?('Workflow') && finding['Workflow'].member?('Status') && finding['Workflow']['Status'] == 'SUPPRESSED'
+        imp = :INFORMATIONAL
+      else
+        # severity is required, but can be either 'label' or 'normalized' internally with 'label' being preferred.  other values can be in here too such as the original severity rating.
+        default = proc { finding['Severity'].key?('Label') ? finding['Severity']['Label'].to_sym : finding['Severity']['Normalized']/100.0 }
+        imp = external_product_handler(finding['ProductArn'], finding, :finding_impact, default)
+      end
+      imp.is_a?(Symbol) ? IMPACT_MAPPING[imp] : imp
+    end
+
+    def desc_tags(data, label)
+      { data: data || NA_STRING, label: label || NA_STRING }
+    end
+
+    def subfindings(finding)
+      subfinding = {}
+
+      statusreason = finding['Compliance']['StatusReasons'].map { |reason| reason.flatten.map { |string| encode(string) } }.flatten.join("\n") if finding.key?('Compliance') && finding['Compliance'].key?('StatusReasons')
+      if finding.key?('Compliance') && finding['Compliance'].key?('Status')
+        case finding['Compliance']['Status']
+        when 'PASSED'
+          subfinding['status'] = 'passed'
+          subfinding['message'] = statusreason if statusreason
+        when 'WARNING'
+          subfinding['status'] = 'skipped'
+          subfinding['skip_message'] = statusreason if statusreason
+        when 'FAILED'
+          subfinding['status'] = 'failed'
+          subfinding['message'] = statusreason if statusreason
+        when 'NOT_AVAILABLE'
+          # primary meaning is that the check could not be performed due to a service outage or API error, but it's also overloaded to mean NOT_APPLICABLE so technically 'skipped' or 'error' could be applicable, but AWS seems to do the equivalent of skipped
+          subfinding['status'] = 'skipped'
+          subfinding['message'] = statusreason if statusreason
+        else
+          subfinding['status'] = 'no_status'
```

Comment:
Can Heimdall process a `no_status` status keyword?

Reply:
Heimdall crashes due to the ControlResultStatus enum not having 'no_status' as an option (https://github.com/mitre/inspecjs/blob/4aa9156d0ee7ca60de599a7f452f814aa4b37cf9/src/generated_parsers/v_1_0/exec-json.ts#L104) even though the SegmentStatus enum does (https://github.com/mitre/inspecjs/blob/master/src/compat_wrappers.ts#L44).  I'm not sure what a 'Segment' is and why it's allowed to not have a status whereas 'ControlResult's cannot.  Can @rbclark comment here?

Reply:
Conclusion was to make it skipped since no_status isn't apparently a valid enum value.